### PR TITLE
#512 write tests administrative panel

### PR DESCRIPTION
**Pull request by romanmyko**
Test:
-api structure
-json structure
-api get requests

### CODE REVIEW

File: administration/tests/test_admin_profiles.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+    def test_get_profiles_not_authorized(self):
+        response = self.client.get(
+            path="/api/admin/profiles/?page=1&page_size=1"
+        )
+        self.assertEqual(status.HTTP_401_UNAUTHORIZED, response.status_code)
+
+    def test_get_profile_id_not_authorized(self):
+        response = self.client.get(
+            path="/api/admin/profiles/" + str(self.user.id) + "/"
```
Comment (BelousSofiya) [inline administration/tests/test_admin_profiles.py]: Please, use f-string or .format. Our expert Ihor prefer .format

File: administration/tests/test_admin_profiles.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    AdminUserFactory,
+    AdminProfileFactory,
+)
+
+from utils.dump_response import dump  # noqa
+
+
+class TestAdminProfilesAPITests(APITestCase):
+    def setUp(self):
+        self.user = AdminUserFactory()
```
Comment (BelousSofiya) [inline administration/tests/test_admin_profiles.py:14]: I think we need some testcases for authenticated but not admin user

File: administration/tests/test_admin_profiles.py [hunk 3]
```diff
@@ -22,27 +22,28 @@ def test_get_profiles_not_authorized(self):
 
     def test_get_profile_id_not_authorized(self):
         response = self.client.get(
-            path="/api/admin/profiles/" + str(self.user.id) + "/"
+            path="/api/admin/profiles/" + str(self.company.id) + "/"
         )
         self.assertEqual(status.HTTP_401_UNAUTHORIZED, response.status_code)
 
-    def test_get_profiles(self):
+    def test_get_profiles_structure_json(self):
```
Comment (BelousSofiya) [inline administration/tests/test_admin_profiles.py:29]: We always have json in response, you don't need to emphasise it in test name, it's better to use '_authenticated_is_not_admin', '_authenticated_is_admin', etc.

File: administration/tests/test_admin_profiles.py [hunk 4]
```diff
@@ -22,27 +22,28 @@ def test_get_profiles_not_authorized(self):
...
         self.client.force_authenticate(self.user)
         response = self.client.get(path="/api/admin/profiles/")
+        auto_data_user = response.json()[0]["person"]
```
Comment (BelousSofiya) [inline administration/tests/test_admin_profiles.py:32]: What is [0]? It's unclear and difficult to understand. What will happen if we change response structure? And auto_data_user['name'] is difficult to read to. Tests have to be as simple and clear as possible. I like your first varient more.

File: administration/tests/test_admin_profiles.py [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+from administration.factories import (
+    AdminUserFactory,
+    AdminProfileFactory,
+)
+
+from utils.dump_response import dump  # noqa
+
+
+class TestAdminProfilesAPITests(APITestCase):
```
Comment (BelousSofiya) [inline administration/tests/test_admin_profiles.py:12]: I am sure we need tests for all BE functional of administration app views. Not only for get operations

File: administration/tests/test_admin_profiles.py (hunk 1 above)
Comment (romanmyko) [reply administration/tests/test_admin_profiles.py]: Done

File: administration/tests/test_admin_profiles.py [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+        )
+        self.assertEqual(status.HTTP_401_UNAUTHORIZED, response.status_code)
+
+    def test_get_profile_id_not_authorized(self):
+        response = self.client.get(
+            path="/api/admin/profiles/" + str(self.user.id) + "/"
+        )
+        self.assertEqual(status.HTTP_401_UNAUTHORIZED, response.status_code)
+
+    def test_get_profiles(self):
```
[the remark this answers is not in the document]
Comment (romanmyko) [reply administration/tests/test_admin_profiles.py]: Done

File: administration/tests/test_admin_profiles.py (hunk 2 above)
Comment (romanmyko) [reply administration/tests/test_admin_profiles.py:35]: Done

File: administration/tests/test_admin_profiles.py (hunk 3 above)
Comment (romanmyko) [reply administration/tests/test_admin_profiles.py:50]: Done

File: administration/tests/test_admin_profiles.py (hunk 5 above)
Comment (romanmyko) [reply administration/tests/test_admin_profiles.py:33]: Done

File: administration/tests/test_admin_profiles.py (hunk 4 above)
Comment (romanmyko) [reply administration/tests/test_admin_profiles.py:53]: It works stably and does not affect the order of starting the generators of the model

File: administration/tests/test_admin_profiles.py [hunk 7]
```diff
@@ -140,13 +140,78 @@ def setUp(self):
         self.profile = AdminProfileFactory()
         self.user = AdminUserFactory()
 
+    def test_Put_Profile(self):
```
Comment (BelousSofiya) [inline administration/tests/test_admin_profiles.py:143]: Change test names to the snake case, please

Comment (romanmyko) [reply administration/tests/test_admin_profiles.py]: done

**Review by BelousSofiya — APPROVED** [review]

File: administration/tests/test_admin_profiles.py [hunk 8]
```diff
@@ -0,0 +1,330 @@
...
+            "official_name": "Test string",
+            "common_info": "Test string",
+            "phone": 123456789012,
+            "edrpou": 12345678,
+            "founded": 2024,
+            "service_info": "Test string",
+            "product_info": "Test string",
+            "address": "Test string",
+            "startup_idea": "Test string",
+            "is_deleted": True,
```
Comment (Lvyshnevska) [inline administration/tests/test_admin_profiles.py:157]: I assume, that we should have separate tests for updating fields and for marking the company as deleted.

**Review by Lvyshnevska — APPROVED** [review]

File: administration/tests/test_admin_user.py [hunk 9]
```diff
@@ -0,0 +1,239 @@
...
+            "is_superuser": True,
+        }
+        response = self.client.put(
+            path=f"/api/admin/users/{self.user.id}/",
+            data=data,
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_put_reverese_bool_user(self):
+        self.client.force_authenticate(self.user)
```
[the remark this answers is not in the document]
Comment (romanmyko) [reply administration/tests/test_admin_user.py:118]: Done rename => test_patch_change_role_user

File: administration/tests/test_admin_profiles.py (hunk 8 above)
Comment (romanmyko) [reply administration/tests/test_admin_profiles.py:157]: There are no requirements for the administrative panel. A comprehensive test has been made

**Pull request merged**